x/farming/types/plan_test.go

```diff
@@ -85,3 +121,42 @@ func TestPrivatePlanFarmingPoolAddress(t *testing.T) {
 	require.Equal(t, testAcc2, sdk.AccAddress(address.Module(types.ModuleName, []byte("PrivatePlan|1|test2"))))
 	require.Equal(t, "cosmos172yhzhxwgwul3s8m6qpgw2ww3auedq4k3dt224543d0sd44fgx4spcjthr", testAcc2.String())
 }
+
+// TODO: needs to cover more cases
+// https://github.com/tendermint/farming/issues/90
+func TestUnpackPlan(t *testing.T) {
+	plan := []types.PlanI{
+		types.NewRatioPlan(
+			types.NewBasePlan(
+				1,
+				"testPlan1",
+				types.PlanTypePrivate,
+				types.PrivatePlanFarmingPoolAddress("farmingPoolAddr1", 1).String(),
+				sdk.AccAddress("terminationAddr1").String(),
+				sdk.NewDecCoins(sdk.DecCoin{Denom: "testFarmStakingCoinDenom", Amount: sdk.MustNewDecFromStr("1.0")}),
+				mustParseRFC3339("2021-08-03T00:00:00Z"),
+				mustParseRFC3339("2021-08-07T00:00:00Z"),
+			),
+			sdk.NewDec(1),
+		),
+	}
+
+	any, err := types.PackPlan(plan[0])
+	require.NoError(t, err)
+
+	planRecord := types.PlanRecord{
+		Plan:             *any,
+		FarmingPoolCoins: sdk.NewCoins(),
+	}
+
+	_, err = types.UnpackPlan(&planRecord.Plan)
+	require.NoError(t, err)
+}
```

Comment:
We should check if this works:
1. Call `PackPlan`
2. Marshal using JSON/protobuf
3. Unmarshal it
4. Call `UnpackPlan`